methods/basic.go

```diff
@@ -15,7 +20,62 @@ func BasicRequestWithBody(c *cli.Context, method string) (string, error) {
 		return "", err
 	}
 
-	var jsonStr = []byte(c.String("body"))
+	var jsonStr []byte
+
+	if c.Bool("editor") {
+		var path string
+		var editor string
+
+		// check OS
+		currentOs := runtime.GOOS
+
+		// based on OS find assign default editor
+		if currentOs == "linux" || currentOs == "darwin" {
+			editor = "nano"
+		} else if currentOs == "windows" {
+			editor = "notepad"
+		}
+
+		path, err = exec.LookPath(editor)
+		if err != nil {
+			return "", fmt.Errorf("Unable to locate %s at %s", editor, path)
+		}
+
+		// create temp file for writing request body
+		tempFile, err := ioutil.TempFile("", "HOPP-CLI-REQUEST*.txt")
```

Comment:
Idiomatically, in most cases the directory prefix for temp files should be passed as `os.TempDir()` instead of an empty string. This is cross-platform.

Reply:
Actually, I checked **ioutil** package doc, for `ioutil.TempFile(dir, prefix)`, it says if **dir** is empty string, the default dir for temp files will be used.

Reply:
Oh, woops! I've always used it explicitly, so I didn't remember that!